content/2022/prepoznavanje-govora.md

```diff
@@ -2,3 +2,239 @@
+
+Hiperravni koje ograničavaju zonu udaljenosti od granice odlučivanja na kojoj klasifikator daje vrednosti čija je apsolutna vrednost manja od 1 nazivaju se noseći vektori. To znači da za svaki podatak koji se nalazi unutar tih vektora ne možemo sa sigurnošću reći kojoj klasi pripada.
+
+![SVM1](static\images\5.png)
+
+Na slici je hiperravan prikazana kao prava u 2D prostoru, dok bi u 3D prostoru to bila ravan i tako dalje.
+
+Za razliku od logističke regresije gde smo sve vrednosti sveli na raspon [0, 1] koristeći sigmoidnu funckiju, ovde sve vrednosti možemo svesti na raspon [-1, 1]. Funkcija gubitka SVM modela je:
+
+$c(x, y, f(x))= \begin{cases}0, & \text { if } y * f(x) \geq 1 \\ 1-y * f(x), & \text { else }\end{cases}$
+
+
+Ako su dobijeni i željeni rezultat istog znaka, vrednost kriterijumske funkcije je jednaka nuli, dok u suprotnom računamo gubitak. Na to moramo dodati i parametar za regularizaciju koji služi da izjednači uticaj maksimizacije granice i minimizacije gubitka.
+
+$$
+\min _w \lambda\|w\|^2+\sum_{i=1}^n\left(1-y_i\left\langle x_i, w\right\rangle\right)_{+}
+$$
+
+Nakon toga možemo izvesti gradijente za ažuriranje vrednosti težina modela:
+
+$\frac{\delta}{\delta w_k} \lambda\|w\|^2=2 \lambda w_k \\$
+
+$\frac{\delta}{\delta w_k}\left(1-y_i\left\langle x_i, w\right\rangle\right)_{+}= \begin{cases}0, & \text { if } y_i\left\langle x_i, w\right\rangle \geq 1 \\ -y_i x_{i k}, & \text { else }\end{cases}$
+
+Težine ažuriramo zavisno od toga da li je naš klasifikator tačno klasifikovao novi podatak ili ne. Ukoliko jeste, ažuriramo samo gradijent regularizacionog parametra:
+
+$$
+w=w-\alpha \cdot(2 \lambda w)
+$$
+
+U suprotnom, ako je model napravio grešku, moramo da uključimo i funkciju gubitka u račun:
+
+$$
+w=w+\alpha \cdot\left(y_i \cdot x_i-2 \lambda w\right)
+$$
+
+##### 6. Konvolucione neuronske mreže
+
+Metoda konvolucionih neuronskih mreža pomaže za klasifikaciju podataka pomoću tehnike dubokog učenja. Neuronske mreže rade po principu čovečjeg mozga (odatle i naziv): dobija određene podatke koji se obično nalaze u formatu baze podataka, obrađuje ih i vraća rezultate. Kontrolom rezultata obrade podataka se ta mreža trenira. Ona uči na svojim greškama i poboljšava rezultate obrade.
+
+Konvoluciona neuronska mreža korišćena u ovom projektu sastoji se iz 5 konvolucionih slojeva, koristi se 4 slojeva sažimanja, kao i 3 potpuno povezana sloja. 
+
+Ceo proces može se svesti na sledeće korake: 
+- Spektrogram se prvo obrađuje konvolucijom i ReLU-om
+- Smanjujemo veličinu obrađene slike pooling slojem
+- Ponavljamo ovaj proces
+
+Konvolucija (po čemu nastaje termin konvolucione neuronske mreže) u obradi slike je operator koji predstavlja sužavanje početne slike množenjem iste određenim filterom. 
```

Comment:
U opštem slučaju ne sužava sliku pa samim tim to i nije informacija koju treba istaći.

Reply:
Rešeno